x-pack/plugins/apm/public/components/app/correlations/context_popover/field_stats_popover.tsx

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiButtonIcon, EuiToolTip } from '@elastic/eui';
+import React, { useCallback, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  FieldPopover,
+  FieldStats,
+  FieldPopoverHeader,
+  FieldStatsServices,
+} from '@kbn/unified-field-list-plugin/public';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { Filter } from '@kbn/es-query';
+import { DataViewField } from '@kbn/data-views-plugin/common';
+import type { FieldTopValuesBucketParams } from '@kbn/unified-field-list-plugin/public';
+import { useApmParams } from '../../../../hooks/use_apm_params';
+import { useApmPluginContext } from '../../../../context/apm_plugin/use_apm_plugin_context';
+import { useFetchParams } from '../use_fetch_params';
+import { ApmPluginStartDeps } from '../../../../plugin';
+import { useApmDataView } from '../../../../hooks/use_apm_data_view';
+import { useTheme } from '../../../../hooks/use_theme';
+
+export type OnAddFilter = ({
+  fieldName,
+  fieldValue,
+  include,
+}: {
+  fieldName: string;
+  fieldValue: string | number;
+  include: boolean;
+}) => void;
+
+const defaultFilters: Filter[] = [];
+
+export function FieldStatsPopover({
+  fieldName,
+  fieldValue,
+  onAddFilter,
+}: {
+  fieldName: string;
+  fieldValue: string | number;
+  onAddFilter: OnAddFilter;
+}) {
+  const {
+    query: { kuery: kueryStr },
+  } = useApmParams('/services/{serviceName}');
+
+  const fieldStatsQuery = {
+    query: kueryStr,
+    language: 'kuery',
+  };
+  const { start, end } = useFetchParams();
+
+  const {
+    data,
+    core: { uiSettings },
+  } = useApmPluginContext();
+  const { dataView } = useApmDataView();
+  const {
+    services: { fieldFormats, charts },
+  } = useKibana<ApmPluginStartDeps>();
+
+  const fieldStatsServices: Partial<FieldStatsServices> = {
```

Comment:
Let's wrap in `useMemo`

Reply:
Updated here https://github.com/elastic/kibana/pull/143416/commits/54769031be5ac55d822d3ce39b7d844cccf52c89